gitpod-io/website
# Minor updates to changelog automation

## Description
Makes sure that PRs without https://github.com/gitpod-io/gitpod/labels/deployed are not present in the changelog.

Also adds partial content replacement so user-added content outside of the `Fixes and improvements` section doesn't get overridden when re-generating.

Also also adds `--dryRun` which makes it possible to not write to the file at all, just output the up-to-date contents to the console. One can use `--onlyPrs` as well to not include the whole file, but only the Dynamic part with the autogenerated list of PRs.

Also also also by default only rewrites the "Fixes and improvements" section, leaving everything else untouched. 

## How to test
1. Generate a changelog

```console
node ./scripts/generate-changelog.js
```

2. Update some contents of the generated file, re-run the script and make sure you still have your changes there (if not in the `AUTOGENERATED_CHANGES` block).

## Release Notes

```release-note
NONE
```

## Documentation



<a href="https://gitpod-staging.com/#https://github.com/gitpod-io/website/pull/2759"><img src="https://gitpod-staging.com/button/open-in-gitpod.svg"/></a>



<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/2759"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
